src/main/java/io/divolte/server/Server.java

```diff
@@ -135,7 +204,9 @@ public void shutdown() {
         }
 
         logger.info("Stopping thread pools.");
-        processingPool.stop();
+        // Stop the mappings before the sinks to ensure work in progress doesn't get stranded.
+        mappingProcessors.values().forEach(ProcessingPool::stop);
+        sinks.values().forEach(ProcessingPool::stop);
 
         logger.info("Closing HDFS filesystem connection.");
```

Comment:
The HDFS client registers a JVM shutdown hook, which we disable again in the HDFS configuration somewhere, because we cannot control the ordering of shutdown hooks, meaning HDFS may close before we flush on shutdown. Hence, we close all HDFS stuff explicitly in our own shutdown sequence.